src/Keymapping-Tools-Spec/KMDescriptionPresenter.class.st

```diff
@@ -56,39 +49,84 @@ KMDescriptionPresenter class >> menuCommandOn: aBuilder [
-	
+
 	^ self new
-		categories: KMRepository default categories keys;
+		addShortcutActivationCmdInstances;
+		addKeymapCategoryInstances;
+		addAllAPIsEntry;
 		open
 ]
 
 { #category : 'api' }
-KMDescriptionPresenter >> categories: aCollectionOfSymbols [
-	categoryList items: (aCollectionOfSymbols collect: [ :e | KMRepository default categoryForName: e ])
+KMDescriptionPresenter >> addAllAPIsEntry [
+	"Set the receiver's category items to a list of categories in aCollectionOfSymbols.
+	We also build an 'All' category including all keymaps in aCollectionOfSymbols"
+
+	| newCategoryItem kmNamedCategories |
+	
+	newCategoryItem := KMCategoryItemPresenter new categoryName: 'All'.
+	kmNamedCategories := KMRepository default categories.
+	kmNamedCategories inject: newCategoryItem into: [ : a : b | a , b ].
+	newCategoryItem addAll: self shortcutActivationCmdInstances.
+
+	categoryList prependCollection: {newCategoryItem}
+]
+
+{ #category : 'api - keymaps' }
+KMDescriptionPresenter >> addKeymapCategoryInstances [
+
+	self kmCategories: KMRepository default categories keys
+]
+
+{ #category : 'api - commander' }
+KMDescriptionPresenter >> addShortcutActivationCmdInstances [
+
+	categoryList appendCollection: self shortcutActivationCategoriesItems.
 ]
 
 { #category : 'initialization' }
 KMDescriptionPresenter >> connectPresenters [
 
 	categoryList
 		transmitTo: shortcutList
-		transform: [ :category | (category model entriesAt: #all) keymaps sorted: [ :keymap | keymap shortcut asString ] ascending ]
+		transform: [ :item | self keymapsAtCategory: item ].
+
+	shortcutList
+		transmitTo: shortcutDetails
+		transform: [ :item | self shortcutDetailsFor: item ]
+		postTransmission: [ :presenter | selectedAction ifNotNil: [ :actionBlock | self updateCodePresenter: actionBlock ] ]
 ]
 
-{ #category : 'initialization - deprecated' }
-KMDescriptionPresenter >> initializeWidgets [
-	categoryList := self newDropList.
-	shortcutList := self newTable.
-	actionBar := self newActionBar.
+{ #category : 'layout' }
+KMDescriptionPresenter >> defaultLayout [
 
-	categoryList display: [ :category | category name ].
+	^ SpPanedLayout newTopToBottom
+		  add: (SpBoxLayout newTopToBottom
+				   spacing: 5;
+				   add: (SpBoxLayout newLeftToRight
+						    add: 'Filter scopes' expand: false;
+						    add: categoryList;
+						    yourself)
+				   expand: false;
+				   add: shortcutList);
+		  add: (SpBoxLayout newTopToBottom
+				   add: shortcutDetails;
+				   add: actionBar withConstraints: [ :c | c height: 25 ]);
+		  yourself
+]
 
-	shortcutList
-		addColumn: (SpStringTableColumn title: 'Shortcut' evaluated: #shortcut);
-		addColumn: (SpStringTableColumn title: 'Name' evaluated: #name);
-		addColumn: (SpStringTableColumn title: 'Description' evaluated: #description);
-		beResizable.
+{ #category : 'callbacks' }
+KMDescriptionPresenter >> highlightColor [
 
+	^ self application configuration isDarkTheme 
```

Comment:
This should probably ask the theme for a color instead so that each theme can configure it